opm/autodiff/BlackoilModelBase_impl.hpp

```diff
@@ -2674,7 +2676,7 @@ namespace detail {
                                                           const ADB& rs,
                                                           const ADB& rv) const
     {
-        const double* rhos = fluid_.surfaceDensity();
+        std::vector<V> rhos = fluid_.surfaceDensity(cells_);
```

Comment:
This does cause more allocation and copying than before, and it should not be necessary (it could be stored in the class). Have you seen (minor) performance reductions with this?